fabric-events-interaction-v0/src/client/java/net/fabricmc/fabric/api/event/client/player/ClientPlayerBlockBreakEvents.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.client.player;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Contains client side events triggered by block breaking.
+ *
+ * <p>For preventing block breaking client side and other purposes, see {@link net.fabricmc.fabric.api.event.player.AttackBlockCallback}.
+ */
+public class ClientPlayerBlockBreakEvents {
+	/**
+	 * Callback after a block is broken.
+	 *
+	 * <p>Only called client side.
+	 */
+	public static final Event<After> AFTER = EventFactory.createArrayBacked(After.class,
+			(listeners) -> (world, player, pos, state, entity) -> {
+				for (After event : listeners) {
+					event.afterBlockBreak(world, player, pos, state, entity);
+				}
+			}
+	);
+
+	@FunctionalInterface
+	public interface After {
+		/**
+		 * Called after a block is successfully broken.
+		 *
+		 * @param world the world where the block was broken
+		 * @param player the player who broke the block
+		 * @param pos the position where the block was broken
+		 * @param state the block state <strong>before</strong> the block was broken
+		 * @param blockEntity the block entity of the broken block, can be {@code null}
+		 */
+		void afterBlockBreak(World world, PlayerEntity player, BlockPos pos, BlockState state, @Nullable BlockEntity blockEntity);
```

Comment:
Why the `BlockEntity` argument, surely if needed a mod can just get it from the world?

Reply:
It’s in the server side api, so I kept the method signature the same.

Reply:
At the point where the server API injects the BE is available, I also think its a lot less likely that you will need access to the BE on the client.

Reply:
Removed :)